File: 1147-flip-columns-for-maximum-number-of-equal-rows/1147-flip-columns-for-maximum-number-of-equal-rows.java
class Solution {
    public int maxEqualRowsAfterFlips(int[][] matrix) {
        Map<String,Integer> freq=new HashMap<>();
        for(int[] row:matrix){
            StringBuilder pattern=new StringBuilder("");
            for(int i=0;i<row.length;i++){
                if(row[0]==row[i]){
                    pattern.append("T");
                }
                else{
                    pattern.append("F");
                }
            }
            freq.put(pattern.toString(),freq.getOrDefault(pattern.toString(),0)+1);
        }
        return Collections.max(freq.values());
    }
}

File: 1147-flip-columns-for-maximum-number-of-equal-rows/1147-flip-columns-for-maximum-number-of-equal-rows.py
class Solution:
    def maxEqualRowsAfterFlips(self, matrix: List[List[int]]) -> int:
        freq=dict()
        for row in matrix:
            pattern=''.join('T' if i==row[0] else 'F' for i in row)
            freq[pattern]=freq.get(pattern,0)+1
        return max(freq.values())
